Document the updated URLs generation (#2267)
platformsh/platformsh-docs: docs/src/configuration/routes/_index.md

#### URLs in non-Production environments

> URLS in non-Production environments follow a different pattern.

Reviewer: ```suggestion
URLs in non-Production environments follow a different pattern.
```

No matter how you have set your default domain (even if you don't have one),
using either the absolute URL or the `{default}` placeholder results in the same URL.

In any case, you get the same URL for a `feature` environment:

```txt
https://feature-t6dnbai-abcdef1234567.us-2.platformsh.site/blog
```

Note that the `example.com` prefix isn't part of the generated URL.

{{< note title="Previous behavior" >}}

> Prior to April 7, 2022, URLs in non-Production environments differed depending on whether or not you used the `{default}` placeholder.

Reviewer: ```suggestion
Before April 7, 2022, URLs in non-Production environments differed depending on whether or not you used the `{default}` placeholder.
```
